### PR TITLE
Image classification : `max train steps`, `one-shot` support

### PR DESCRIPTION
**Pull request by rahul-tuli**
The goal of this PR is to add support for 
- `--max-train-steps` 
- `--max-eval-steps`
- `--one-shot`

Arguments for `sparseml.image_classification.train` script. the idea being such args will allow testing shorter runs and help provide stronger validation for shipped models. Noting `max-train-steps` and `max-eval-steps` are run per epoch and no training take place when `one-shot` argument is passed

The PR has been tested locally with the following:

1) Max train steps = 5, Max eval steps = 4
```bash
sparseml.image_classification.train \
    --dataset-path ./datasets/imagenette \
    --max-train-steps 5 \
    --max-eval-steps 4 \
    --arch-key resnet50 \
    --recipe-path zoo:cv/classification/resnet_v1-50/pytorch/sparseml/imagenette/pruned-conservative?recipe_type=original \
    --checkpoint-path zoo \
    --train_batch_size 32 \
    --test_batch_size 32 \
    --dataset imagenette
```  
Output:
```bash
Test epoch -1: 100%|██████████████████████████████| 4/4 [00:00<00:00,  5.48it/s]
2022-05-10 16:15:06 __main__     INFO     Starting training from epoch 0
2022-05-10 16:15:06 __main__     INFO     Starting training from epoch 0
/XXXXXXX/projects/sparseml/src/sparseml/pytorch/optim/optimizer.py:186: UserWarning: ScheduledOptimizer is deprecated and will be deleted in the future. adjust_current_step is no longer supported. Please replace with manager.initialize and manager.modify
  warnings.warn(
Train epoch 0: 100%|██████████████████████████████| 5/5 [00:01<00:00,  4.51it/s]
2022-05-10 16:15:08 __main__     INFO     
Epoch 0 training results: ModuleRunResults(__loss__=0.14738957583904266, top1acc=97.39583587646484, top5acc=99.47916412353516)
2022-05-10 16:15:08 __main__     INFO     
Epoch 0 training results: ModuleRunResults(__loss__=0.14738957583904266, top1acc=97.39583587646484, top5acc=99.47916412353516)
Test epoch 0: 100%|███████████████████████████████| 4/4 [00:00<00:00,  7.98it/s]
Train epoch 1: 100%|██████████████████████████████| 5/5 [00:01<00:00,  4.17it/s]
2022-05-10 16:15:09 __main__     INFO     
Epoch 1 training results: ModuleRunResults(__loss__=0.06656134128570557, top1acc=97.91666412353516, top5acc=100.0)
2022-05-10 16:15:09 __main__     INFO     
Epoch 1 training results: ModuleRunResults(__loss__=0.06656134128570557, top1acc=97.91666412353516, top5acc=100.0)
Test epoch 1: 100%|███████████████████████████████| 4/4 [00:00<00:00,  7.55it/s]
Saving model for epoch 1 and top-1 accuracy 96.875 to /XXXXXXX/projects/sparseml/src/sparseml/pytorch/image_classification/pytorch_vision/resnet50_imagenette__09 for checkpoint-best
Train epoch 2: 100%|██████████████████████████████| 5/5 [00:01<00:00,  4.22it/s]
2022-05-10 16:15:11 __main__     INFO     
Epoch 2 training results: ModuleRunResults(__loss__=0.10613694787025452, top1acc=96.875, top5acc=100.0)
2022-05-10 16:15:11 __main__     INFO     
Epoch 2 training results: ModuleRunResults(__loss__=0.10613694787025452, top1acc=96.875, top5acc=100.0)
Test epoch 2: 100%|███████████████████████████████| 4/4 [00:00<00:00,  7.27it/s]
Train epoch 3: 100%|██████████████████████████████| 5/5 [00:01<00:00,  4.18it/s]
2022-05-10 16:15:13 __main__     INFO     
Epoch 3 training results: ModuleRunResults(__loss__=0.15126116573810577, top1acc=95.83333587646484, top5acc=100.0)
2022-05-10 16:15:13 __main__     INFO     
Epoch 3 training results: ModuleRunResults(__loss__=0.15126116573810577, top1acc=95.83333587646484, top5acc=100.0)
Test epoch 3: 100%|███████████████████████████████| 4/4 [00:00<00:00,  7.39it/s]
Train epoch 4: 100%|██████████████████████████████| 5/5 [00:01<00:00,  4.21it/s]
2022-05-10 16:15:15 __main__     INFO     
Epoch 4 training results: ModuleRunResults(__loss__=0.414724200963974, top1acc=90.625, top5acc=97.91666412353516)
2022-05-10 16:15:15 __main__     INFO     
Epoch 4 training results: ModuleRunResults(__loss__=0.414724200963974, top1acc=90.625, top5acc=97.91666412353516)
Test epoch 4: 100%|███████████████████████████████| 4/4 [00:00<00:00,  7.54it/s]
Train epoch 5: 100%|██████████████████████████████| 5/5 [00:01<00:00,  4.31it/s]
2022-05-10 16:15:17 __main__     INFO     
Epoch 5 training results: ModuleRunResults(__loss__=0.37264159321784973, top1acc=93.75, top5acc=100.0)
2022-05-10 16:15:17 __main__     INFO     
Epoch 5 training results: ModuleRunResults(__loss__=0.37264159321784973, top1acc=93.75, top5acc=100.0)
Test epoch 5: 100%|███████████████████████████████| 4/4 [00:00<00:00,  7.24it/s]
Train epoch 6: 100%|██████████████████████████████| 5/5 [00:01<00:00,  4.21it/s]
2022-05-10 16:15:19 __main__     INFO     
Epoch 6 training results: ModuleRunResults(__loss__=0.2325858324766159, top1acc=97.91666412353516, top5acc=99.47916412353516)
2022-05-10 16:15:19 __main__     INFO     
Epoch 6 training results: ModuleRunResults(__loss__=0.2325858324766159, top1acc=97.91666412353516, top5acc=99.47916412353516)
Test epoch 6: 100%|███████████████████████████████| 4/4 [00:00<00:00,  7.21it/s]
Train epoch 7: 100%|██████████████████████████████| 5/5 [00:01<00:00,  4.86it/s]
2022-05-10 16:15:20 __main__     INFO     
Epoch 7 training results: ModuleRunResults(__loss__=0.22111153602600098, top1acc=93.75, top5acc=99.47916412353516)
2022-05-10 16:15:20 __main__     INFO     
Epoch 7 training results: ModuleRunResults(__loss__=0.22111153602600098, top1acc=93.75, top5acc=99.47916412353516)
Test epoch 7: 100%|███████████████████████████████| 4/4 [00:00<00:00,  7.43it/s]
Train epoch 8: 100%|██████████████████████████████| 5/5 [00:01<00:00,  4.89it/s]
2022-05-10 16:15:22 __main__     INFO     
Epoch 8 training results: ModuleRunResults(__loss__=0.11022845655679703, top1acc=98.4375, top5acc=100.0)
2022-05-10 16:15:22 __main__     INFO     
Epoch 8 training results: ModuleRunResults(__loss__=0.11022845655679703, top1acc=98.4375, top5acc=100.0)
Test epoch 8: 100%|███████████████████████████████| 4/4 [00:00<00:00,  7.34it/s]
Train epoch 9: 100%|██████████████████████████████| 5/5 [00:01<00:00,  4.83it/s]
2022-05-10 16:15:24 __main__     INFO     
Epoch 9 training results: ModuleRunResults(__loss__=0.14420939981937408, top1acc=96.875, top5acc=99.47916412353516)
2022-05-10 16:15:24 __main__     INFO     
Epoch 9 training results: ModuleRunResults(__loss__=0.14420939981937408, top1acc=96.875, top5acc=99.47916412353516)
Test epoch 9: 100%|███████████████████████████████| 4/4 [00:00<00:00,  7.38it/s]
2022-05-10 16:15:24 __main__     INFO     completed...
2022-05-10 16:15:24 __main__     INFO     completed...
Saving model for epoch 9 and top-1 accuracy 90.625 to /XXXXXXX/projects/sparseml/src/sparseml/pytorch/image_classification/pytorch_vision/resnet50_imagenette__09 for model
2022-05-10 16:15:25 __main__     INFO     layer sparsities:
2022-05-10 16:15:25 __main__     INFO     layer sparsities:
2022-05-10 16:15:25 __main__     INFO     input.conv.weight: 0.0000
2022-05-10 16:15:25 __main__     INFO     input.conv.weight: 0.0000
2022-05-10 16:15:25 __main__     INFO     sections.0.0.conv1.weight: 0.8000
2022-05-10 16:15:25 __main__     INFO     sections.0.0.conv1.weight: 0.8000
2022-05-10 16:15:25 __main__     INFO     sections.0.0.conv2.weight: 0.8000
2022-05-10 16:15:25 __main__     INFO     sections.0.0.conv2.weight: 0.8000
2022-05-10 16:15:25 __main__     INFO     sections.0.0.conv3.weight: 0.8000
2022-05-10 16:15:25 __main__     INFO     sections.0.0.conv3.weight: 0.8000
2022-05-10 16:15:25 __main__     INFO     sections.0.0.identity.conv.weight: 0.8000
2022-05-10 16:15:25 __main__     INFO     sections.0.0.identity.conv.weight: 0.8000
2022-05-10 16:15:25 __main__     INFO     sections.0.1.conv1.weight: 0.8000
2022-05-10 16:15:25 __main__     INFO     sections.0.1.conv1.weight: 0.8000
2022-05-10 16:15:25 __main__     INFO     sections.0.1.conv2.weight: 0.8500
2022-05-10 16:15:25 __main__     INFO     sections.0.1.conv2.weight: 0.8500
2022-05-10 16:15:25 __main__     INFO     sections.0.1.conv3.weight: 0.8000
2022-05-10 16:15:25 __main__     INFO     sections.0.1.conv3.weight: 0.8000
2022-05-10 16:15:25 __main__     INFO     sections.0.2.conv1.weight: 0.8000
2022-05-10 16:15:25 __main__     INFO     sections.0.2.conv1.weight: 0.8000
2022-05-10 16:15:25 __main__     INFO     sections.0.2.conv2.weight: 0.8500
2022-05-10 16:15:25 __main__     INFO     sections.0.2.conv2.weight: 0.8500
2022-05-10 16:15:25 __main__     INFO     sections.0.2.conv3.weight: 0.8000
2022-05-10 16:15:25 __main__     INFO     sections.0.2.conv3.weight: 0.8000
2022-05-10 16:15:25 __main__     INFO     sections.1.0.conv1.weight: 0.8000
2022-05-10 16:15:25 __main__     INFO     sections.1.0.conv1.weight: 0.8000
2022-05-10 16:15:25 __main__     INFO     sections.1.0.conv2.weight: 0.9000
2022-05-10 16:15:25 __main__     INFO     sections.1.0.conv2.weight: 0.9000
2022-05-10 16:15:25 __main__     INFO     sections.1.0.conv3.weight: 0.8000
2022-05-10 16:15:25 __main__     INFO     sections.1.0.conv3.weight: 0.8000
2022-05-10 16:15:25 __main__     INFO     sections.1.0.identity.conv.weight: 0.8500
2022-05-10 16:15:25 __main__     INFO     sections.1.0.identity.conv.weight: 0.8500
2022-05-10 16:15:25 __main__     INFO     sections.1.1.conv1.weight: 0.8500
2022-05-10 16:15:25 __main__     INFO     sections.1.1.conv1.weight: 0.8500
2022-05-10 16:15:25 __main__     INFO     sections.1.1.conv2.weight: 0.8500
2022-05-10 16:15:25 __main__     INFO     sections.1.1.conv2.weight: 0.8500
2022-05-10 16:15:25 __main__     INFO     sections.1.1.conv3.weight: 0.8500
2022-05-10 16:15:25 __main__     INFO     sections.1.1.conv3.weight: 0.8500
2022-05-10 16:15:25 __main__     INFO     sections.1.2.conv1.weight: 0.8500
2022-05-10 16:15:25 __main__     INFO     sections.1.2.conv1.weight: 0.8500
2022-05-10 16:15:25 __main__     INFO     sections.1.2.conv2.weight: 0.8500
2022-05-10 16:15:25 __main__     INFO     sections.1.2.conv2.weight: 0.8500
2022-05-10 16:15:25 __main__     INFO     sections.1.2.conv3.weight: 0.8000
2022-05-10 16:15:25 __main__     INFO     sections.1.2.conv3.weight: 0.8000
2022-05-10 16:15:25 __main__     INFO     sections.1.3.conv1.weight: 0.8000
2022-05-10 16:15:25 __main__     INFO     sections.1.3.conv1.weight: 0.8000
2022-05-10 16:15:25 __main__     INFO     sections.1.3.conv2.weight: 0.8500
2022-05-10 16:15:25 __main__     INFO     sections.1.3.conv2.weight: 0.8500
2022-05-10 16:15:25 __main__     INFO     sections.1.3.conv3.weight: 0.8500
2022-05-10 16:15:25 __main__     INFO     sections.1.3.conv3.weight: 0.8500
2022-05-10 16:15:25 __main__     INFO     sections.2.0.conv1.weight: 0.8000
2022-05-10 16:15:25 __main__     INFO     sections.2.0.conv1.weight: 0.8000
2022-05-10 16:15:25 __main__     INFO     sections.2.0.conv2.weight: 0.9000
2022-05-10 16:15:25 __main__     INFO     sections.2.0.conv2.weight: 0.9000
2022-05-10 16:15:25 __main__     INFO     sections.2.0.conv3.weight: 0.8500
2022-05-10 16:15:25 __main__     INFO     sections.2.0.conv3.weight: 0.8500
2022-05-10 16:15:25 __main__     INFO     sections.2.0.identity.conv.weight: 0.8500
2022-05-10 16:15:25 __main__     INFO     sections.2.0.identity.conv.weight: 0.8500
2022-05-10 16:15:25 __main__     INFO     sections.2.1.conv1.weight: 0.8500
2022-05-10 16:15:25 __main__     INFO     sections.2.1.conv1.weight: 0.8500
2022-05-10 16:15:25 __main__     INFO     sections.2.1.conv2.weight: 0.9000
2022-05-10 16:15:25 __main__     INFO     sections.2.1.conv2.weight: 0.9000
2022-05-10 16:15:25 __main__     INFO     sections.2.1.conv3.weight: 0.8500
2022-05-10 16:15:25 __main__     INFO     sections.2.1.conv3.weight: 0.8500
2022-05-10 16:15:25 __main__     INFO     sections.2.2.conv1.weight: 0.8500
2022-05-10 16:15:25 __main__     INFO     sections.2.2.conv1.weight: 0.8500
2022-05-10 16:15:25 __main__     INFO     sections.2.2.conv2.weight: 0.9000
2022-05-10 16:15:25 __main__     INFO     sections.2.2.conv2.weight: 0.9000
2022-05-10 16:15:25 __main__     INFO     sections.2.2.conv3.weight: 0.8500
2022-05-10 16:15:25 __main__     INFO     sections.2.2.conv3.weight: 0.8500
2022-05-10 16:15:25 __main__     INFO     sections.2.3.conv1.weight: 0.8500
2022-05-10 16:15:25 __main__     INFO     sections.2.3.conv1.weight: 0.8500
2022-05-10 16:15:25 __main__     INFO     sections.2.3.conv2.weight: 0.9000
2022-05-10 16:15:25 __main__     INFO     sections.2.3.conv2.weight: 0.9000
2022-05-10 16:15:25 __main__     INFO     sections.2.3.conv3.weight: 0.8500
2022-05-10 16:15:25 __main__     INFO     sections.2.3.conv3.weight: 0.8500
2022-05-10 16:15:25 __main__     INFO     sections.2.4.conv1.weight: 0.8500
2022-05-10 16:15:25 __main__     INFO     sections.2.4.conv1.weight: 0.8500
2022-05-10 16:15:25 __main__     INFO     sections.2.4.conv2.weight: 0.9000
2022-05-10 16:15:25 __main__     INFO     sections.2.4.conv2.weight: 0.9000
2022-05-10 16:15:25 __main__     INFO     sections.2.4.conv3.weight: 0.8500
2022-05-10 16:15:25 __main__     INFO     sections.2.4.conv3.weight: 0.8500
2022-05-10 16:15:25 __main__     INFO     sections.2.5.conv1.weight: 0.8500
2022-05-10 16:15:25 __main__     INFO     sections.2.5.conv1.weight: 0.8500
2022-05-10 16:15:25 __main__     INFO     sections.2.5.conv2.weight: 0.9000
2022-05-10 16:15:25 __main__     INFO     sections.2.5.conv2.weight: 0.9000
2022-05-10 16:15:25 __main__     INFO     sections.2.5.conv3.weight: 0.8500
2022-05-10 16:15:25 __main__     INFO     sections.2.5.conv3.weight: 0.8500
2022-05-10 16:15:25 __main__     INFO     sections.3.0.conv1.weight: 0.8000
2022-05-10 16:15:25 __main__     INFO     sections.3.0.conv1.weight: 0.8000
2022-05-10 16:15:25 __main__     INFO     sections.3.0.conv2.weight: 0.9000
2022-05-10 16:15:25 __main__     INFO     sections.3.0.conv2.weight: 0.9000
2022-05-10 16:15:25 __main__     INFO     sections.3.0.conv3.weight: 0.9000
2022-05-10 16:15:25 __main__     INFO     sections.3.0.conv3.weight: 0.9000
2022-05-10 16:15:25 __main__     INFO     sections.3.0.identity.conv.weight: 0.9000
2022-05-10 16:15:25 __main__     INFO     sections.3.0.identity.conv.weight: 0.9000
2022-05-10 16:15:25 __main__     INFO     sections.3.1.conv1.weight: 0.8500
2022-05-10 16:15:25 __main__     INFO     sections.3.1.conv1.weight: 0.8500
2022-05-10 16:15:25 __main__     INFO     sections.3.1.conv2.weight: 0.9000
2022-05-10 16:15:25 __main__     INFO     sections.3.1.conv2.weight: 0.9000
2022-05-10 16:15:25 __main__     INFO     sections.3.1.conv3.weight: 0.9000
2022-05-10 16:15:25 __main__     INFO     sections.3.1.conv3.weight: 0.9000
2022-05-10 16:15:25 __main__     INFO     sections.3.2.conv1.weight: 0.8500
2022-05-10 16:15:25 __main__     INFO     sections.3.2.conv1.weight: 0.8500
2022-05-10 16:15:25 __main__     INFO     sections.3.2.conv2.weight: 0.9000
2022-05-10 16:15:25 __main__     INFO     sections.3.2.conv2.weight: 0.9000
2022-05-10 16:15:25 __main__     INFO     sections.3.2.conv3.weight: 0.9000
2022-05-10 16:15:25 __main__     INFO     sections.3.2.conv3.weight: 0.9000
2022-05-10 16:15:25 __main__     INFO     classifier.fc.weight: 0.8000
2022-05-10 16:15:25 __main__     INFO     classifier.fc.weight: 0.8000

Process finished with exit code 0

```

2) With one shot using the following command
```bash
sparseml.image_classification.train \
    --train_batch_size 32 \
    --test_batch_size 32 \
    --dataset imagenette \
    --dataset-path ./datasets/imagenette \
    --arch-key resnet50 \
    --recipe-path zoo:cv/classification/resnet_v1-50/pytorch/sparseml/imagenette/pruned-conservative?recipe_type=original \
    --checkpoint-path zoo \
    --one-shot
```

Output:

```bash
2022-05-11 13:19:01 __main__     INFO     running on device cuda
2022-05-11 13:19:01 __main__     INFO     running on device cuda
2022-05-11 13:19:02 sparseml.optim.manager INFO     Created recipe manager with metadata: {
 "__metadata__": null
}
2022-05-11 13:19:02 __main__     INFO     completed...
2022-05-11 13:19:02 __main__     INFO     completed...
Saving model for epoch -1 to /XXXXXXX//src/sparseml/pytorch/image_classification/pytorch_vision/resnet50_imagenette__10 for model-one-shot
2022-05-11 13:19:02 __main__     INFO     layer sparsities:
2022-05-11 13:19:02 __main__     INFO     layer sparsities:
2022-05-11 13:19:02 __main__     INFO     input.conv.weight: 0.0000
2022-05-11 13:19:02 __main__     INFO     input.conv.weight: 0.0000
2022-05-11 13:19:02 __main__     INFO     sections.0.0.conv1.weight: 0.8000
2022-05-11 13:19:02 __main__     INFO     sections.0.0.conv1.weight: 0.8000
2022-05-11 13:19:02 __main__     INFO     sections.0.0.conv2.weight: 0.8000
2022-05-11 13:19:02 __main__     INFO     sections.0.0.conv2.weight: 0.8000
2022-05-11 13:19:02 __main__     INFO     sections.0.0.conv3.weight: 0.8000
2022-05-11 13:19:02 __main__     INFO     sections.0.0.conv3.weight: 0.8000
2022-05-11 13:19:02 __main__     INFO     sections.0.0.identity.conv.weight: 0.8000
2022-05-11 13:19:02 __main__     INFO     sections.0.0.identity.conv.weight: 0.8000
2022-05-11 13:19:02 __main__     INFO     sections.0.1.conv1.weight: 0.8000
2022-05-11 13:19:02 __main__     INFO     sections.0.1.conv1.weight: 0.8000
2022-05-11 13:19:02 __main__     INFO     sections.0.1.conv2.weight: 0.8500
2022-05-11 13:19:02 __main__     INFO     sections.0.1.conv2.weight: 0.8500
2022-05-11 13:19:02 __main__     INFO     sections.0.1.conv3.weight: 0.8000
2022-05-11 13:19:02 __main__     INFO     sections.0.1.conv3.weight: 0.8000
2022-05-11 13:19:02 __main__     INFO     sections.0.2.conv1.weight: 0.8000
2022-05-11 13:19:02 __main__     INFO     sections.0.2.conv1.weight: 0.8000
2022-05-11 13:19:02 __main__     INFO     sections.0.2.conv2.weight: 0.8500
2022-05-11 13:19:02 __main__     INFO     sections.0.2.conv2.weight: 0.8500
2022-05-11 13:19:02 __main__     INFO     sections.0.2.conv3.weight: 0.8000
2022-05-11 13:19:02 __main__     INFO     sections.0.2.conv3.weight: 0.8000
2022-05-11 13:19:02 __main__     INFO     sections.1.0.conv1.weight: 0.8000
2022-05-11 13:19:02 __main__     INFO     sections.1.0.conv1.weight: 0.8000
2022-05-11 13:19:02 __main__     INFO     sections.1.0.conv2.weight: 0.9000
2022-05-11 13:19:02 __main__     INFO     sections.1.0.conv2.weight: 0.9000
2022-05-11 13:19:02 __main__     INFO     sections.1.0.conv3.weight: 0.8000
2022-05-11 13:19:02 __main__     INFO     sections.1.0.conv3.weight: 0.8000
2022-05-11 13:19:02 __main__     INFO     sections.1.0.identity.conv.weight: 0.8500
2022-05-11 13:19:02 __main__     INFO     sections.1.0.identity.conv.weight: 0.8500
2022-05-11 13:19:02 __main__     INFO     sections.1.1.conv1.weight: 0.8500
2022-05-11 13:19:02 __main__     INFO     sections.1.1.conv1.weight: 0.8500
2022-05-11 13:19:02 __main__     INFO     sections.1.1.conv2.weight: 0.8500
2022-05-11 13:19:02 __main__     INFO     sections.1.1.conv2.weight: 0.8500
2022-05-11 13:19:02 __main__     INFO     sections.1.1.conv3.weight: 0.8500
2022-05-11 13:19:02 __main__     INFO     sections.1.1.conv3.weight: 0.8500
2022-05-11 13:19:02 __main__     INFO     sections.1.2.conv1.weight: 0.8500
2022-05-11 13:19:02 __main__     INFO     sections.1.2.conv1.weight: 0.8500
2022-05-11 13:19:02 __main__     INFO     sections.1.2.conv2.weight: 0.8500
2022-05-11 13:19:02 __main__     INFO     sections.1.2.conv2.weight: 0.8500
2022-05-11 13:19:02 __main__     INFO     sections.1.2.conv3.weight: 0.8000
2022-05-11 13:19:02 __main__     INFO     sections.1.2.conv3.weight: 0.8000
2022-05-11 13:19:02 __main__     INFO     sections.1.3.conv1.weight: 0.8000
2022-05-11 13:19:02 __main__     INFO     sections.1.3.conv1.weight: 0.8000
2022-05-11 13:19:02 __main__     INFO     sections.1.3.conv2.weight: 0.8500
2022-05-11 13:19:02 __main__     INFO     sections.1.3.conv2.weight: 0.8500
2022-05-11 13:19:02 __main__     INFO     sections.1.3.conv3.weight: 0.8500
2022-05-11 13:19:02 __main__     INFO     sections.1.3.conv3.weight: 0.8500
2022-05-11 13:19:02 __main__     INFO     sections.2.0.conv1.weight: 0.8000
2022-05-11 13:19:02 __main__     INFO     sections.2.0.conv1.weight: 0.8000
2022-05-11 13:19:02 __main__     INFO     sections.2.0.conv2.weight: 0.9000
2022-05-11 13:19:02 __main__     INFO     sections.2.0.conv2.weight: 0.9000
2022-05-11 13:19:02 __main__     INFO     sections.2.0.conv3.weight: 0.8500
2022-05-11 13:19:02 __main__     INFO     sections.2.0.conv3.weight: 0.8500
2022-05-11 13:19:02 __main__     INFO     sections.2.0.identity.conv.weight: 0.8500
2022-05-11 13:19:02 __main__     INFO     sections.2.0.identity.conv.weight: 0.8500
2022-05-11 13:19:02 __main__     INFO     sections.2.1.conv1.weight: 0.8500
2022-05-11 13:19:02 __main__     INFO     sections.2.1.conv1.weight: 0.8500
2022-05-11 13:19:02 __main__     INFO     sections.2.1.conv2.weight: 0.9000
2022-05-11 13:19:02 __main__     INFO     sections.2.1.conv2.weight: 0.9000
2022-05-11 13:19:02 __main__     INFO     sections.2.1.conv3.weight: 0.8500
2022-05-11 13:19:02 __main__     INFO     sections.2.1.conv3.weight: 0.8500
2022-05-11 13:19:02 __main__     INFO     sections.2.2.conv1.weight: 0.8500
2022-05-11 13:19:02 __main__     INFO     sections.2.2.conv1.weight: 0.8500
2022-05-11 13:19:02 __main__     INFO     sections.2.2.conv2.weight: 0.9000
2022-05-11 13:19:02 __main__     INFO     sections.2.2.conv2.weight: 0.9000
2022-05-11 13:19:02 __main__     INFO     sections.2.2.conv3.weight: 0.8500
2022-05-11 13:19:02 __main__     INFO     sections.2.2.conv3.weight: 0.8500
2022-05-11 13:19:02 __main__     INFO     sections.2.3.conv1.weight: 0.8500
2022-05-11 13:19:02 __main__     INFO     sections.2.3.conv1.weight: 0.8500
2022-05-11 13:19:02 __main__     INFO     sections.2.3.conv2.weight: 0.9000
2022-05-11 13:19:02 __main__     INFO     sections.2.3.conv2.weight: 0.9000
2022-05-11 13:19:02 __main__     INFO     sections.2.3.conv3.weight: 0.8500
2022-05-11 13:19:02 __main__     INFO     sections.2.3.conv3.weight: 0.8500
2022-05-11 13:19:02 __main__     INFO     sections.2.4.conv1.weight: 0.8500
2022-05-11 13:19:02 __main__     INFO     sections.2.4.conv1.weight: 0.8500
2022-05-11 13:19:02 __main__     INFO     sections.2.4.conv2.weight: 0.9000
2022-05-11 13:19:02 __main__     INFO     sections.2.4.conv2.weight: 0.9000
2022-05-11 13:19:02 __main__     INFO     sections.2.4.conv3.weight: 0.8500
2022-05-11 13:19:02 __main__     INFO     sections.2.4.conv3.weight: 0.8500
2022-05-11 13:19:02 __main__     INFO     sections.2.5.conv1.weight: 0.8500
2022-05-11 13:19:02 __main__     INFO     sections.2.5.conv1.weight: 0.8500
2022-05-11 13:19:02 __main__     INFO     sections.2.5.conv2.weight: 0.9000
2022-05-11 13:19:02 __main__     INFO     sections.2.5.conv2.weight: 0.9000
2022-05-11 13:19:02 __main__     INFO     sections.2.5.conv3.weight: 0.8500
2022-05-11 13:19:02 __main__     INFO     sections.2.5.conv3.weight: 0.8500
2022-05-11 13:19:02 __main__     INFO     sections.3.0.conv1.weight: 0.8000
2022-05-11 13:19:02 __main__     INFO     sections.3.0.conv1.weight: 0.8000
2022-05-11 13:19:02 __main__     INFO     sections.3.0.conv2.weight: 0.9000
2022-05-11 13:19:02 __main__     INFO     sections.3.0.conv2.weight: 0.9000
2022-05-11 13:19:02 __main__     INFO     sections.3.0.conv3.weight: 0.9000
2022-05-11 13:19:02 __main__     INFO     sections.3.0.conv3.weight: 0.9000
2022-05-11 13:19:02 __main__     INFO     sections.3.0.identity.conv.weight: 0.9000
2022-05-11 13:19:02 __main__     INFO     sections.3.0.identity.conv.weight: 0.9000
2022-05-11 13:19:02 __main__     INFO     sections.3.1.conv1.weight: 0.8500
2022-05-11 13:19:02 __main__     INFO     sections.3.1.conv1.weight: 0.8500
2022-05-11 13:19:02 __main__     INFO     sections.3.1.conv2.weight: 0.9000
2022-05-11 13:19:02 __main__     INFO     sections.3.1.conv2.weight: 0.9000
2022-05-11 13:19:02 __main__     INFO     sections.3.1.conv3.weight: 0.9000
2022-05-11 13:19:02 __main__     INFO     sections.3.1.conv3.weight: 0.9000
2022-05-11 13:19:02 __main__     INFO     sections.3.2.conv1.weight: 0.8500
2022-05-11 13:19:02 __main__     INFO     sections.3.2.conv1.weight: 0.8500
2022-05-11 13:19:02 __main__     INFO     sections.3.2.conv2.weight: 0.9000
2022-05-11 13:19:02 __main__     INFO     sections.3.2.conv2.weight: 0.9000
2022-05-11 13:19:02 __main__     INFO     sections.3.2.conv3.weight: 0.9000
2022-05-11 13:19:02 __main__     INFO     sections.3.2.conv3.weight: 0.9000
2022-05-11 13:19:02 __main__     INFO     classifier.fc.weight: 0.8000
2022-05-11 13:19:02 __main__     INFO     classifier.fc.weight: 0.8000

Process finished with exit code 0

```

Also noting for easier reviews: Changes for one-shot support have been reviewed from PR #779 and have been merged into this branch (by mistake)

### CODE REVIEW

File: src/sparseml/pytorch/image_classification/train.py [hunk 1]
```diff
@@ -627,11 +656,12 @@ def train(
     :param save_epochs: The epochs to save checkpoints for
     :param rank: The rank of the process
     """
+    max_steps = debug_steps if debug_steps > 0 else trainer.max_train_steps
 
     # Baseline eval run
     trainer.run_one_epoch(
         mode="validation",
-        max_steps=debug_steps,
+        max_steps=max_steps,
```
Comment (bfineran) [inline src/sparseml/pytorch/image_classification/train.py:664]: should have a separate arg for max eval steps

File: src/sparseml/pytorch/image_classification/utils/helpers.py [hunk 2]
```diff
@@ -286,7 +287,7 @@ def create_model(
         else:
             model, arch_key = result
 
-        return model, arch_key
+        return model, arch_key, checkpoint_path
```
Comment (bfineran) [inline src/sparseml/pytorch/image_classification/utils/helpers.py:290]: where is the checkpoint path used in the main script?

**Review by bfineran — CHANGES_REQUESTED** [review]

File: src/sparseml/pytorch/image_classification/utils/helpers.py (hunk 2 above)
Comment (rahul-tuli) [reply src/sparseml/pytorch/image_classification/utils/helpers.py:290]: It's used while initializing the `trainer`, more specifically it is used to read the recipe from checkpoint in `_setup_checkpoint_manager(...)` method [here](https://github.com/neuralmagic/sparseml/blob/d1584610ab6fc28782eb668102ea81577fe6f6dd/src/sparseml/pytorch/image_classification/utils/trainer.py#L307)

**Review by bfineran — DISMISSED** [review]

**Review by bfineran — COMMENTED** [review]
looks like rebase isn't clean - one_shot is still in the diff

**Comment by rahul-tuli** [comment]
> looks like rebase isn't clean - one_shot is still in the diff

#779 was accidentally merged into this branch

**Comment by bfineran** [comment]
noted, LGTM then

**Review by bfineran — DISMISSED** [review]

**Review by dbogunowicz — CHANGES_REQUESTED** [review]

File: src/sparseml/pytorch/image_classification/train.py [hunk 3]
```diff
@@ -135,6 +132,19 @@
...
+  --max-train-steps, --max_train_steps INTEGER
+                                  The maximum number of training steps to run
+                                  per epoch. If negative, will run for the
+                                  entire dataset. Note if max_eval_steps is
+                                  positive then this is ignored  [default: -1]
+  --max-eval-steps, --max_eval_steps INTEGER
+                                  The maximum number of eval steps to run per
+                                  epoch. If negative, will run for the entire
+                                  dataset. Note if max_eval_steps is positive
+                                  then this is ignored  [default: -1]
```
[the remark this answers is not in the document]
Comment (rahul-tuli) [reply src/sparseml/pytorch/image_classification/train.py:144]: Good catch, thanks! updated

**Review by dbogunowicz — APPROVED** [review]

**Review by bfineran — APPROVED** [review]

**Pull request merged**